x-pack/plugins/ml/server/models/data_frame_analytics/validation.ts

```diff
@@ -69,6 +78,67 @@ const analysisFieldsWarningMessage = {
   heading: analysisFieldsHeading,
 };
 
+function getRegressionAndClassificationMessage(
+  analysisConfig: AnalysisConfig,
+  analysisType: string,
+  totalDocs: number,
+  depVarCardinality: number | undefined
+) {
+  const messages = [];
+  if (isRegressionAnalysis(analysisConfig) || isClassificationAnalysis(analysisConfig)) {
+    const trainingPercent = analysisConfig[analysisType].training_percent;
+    const featureImportance = analysisConfig[analysisType].num_top_feature_importance_values;
+    // @ts-ignore num_top_classes doesn't exist on Regression type
+    const topClasses: number | undefined = analysisConfig[analysisType].num_top_classes;
+
+    if (trainingPercent) {
+      const trainingDocs = totalDocs * (trainingPercent / 100);
+      const trainingPercentMessage = getTrainingPercentMessage(trainingDocs);
+      if (trainingPercentMessage) {
+        messages.push(trainingPercentMessage);
+      }
+    }
+
+    if (featureImportance && featureImportance > 0) {
+      messages.push({
+        id: 'feature_importance',
+        text: i18n.translate(
+          'xpack.ml.models.dfaValidation.messages.featureImportanceWarningMessage',
+          {
+            defaultMessage:
+              'Enabling feature importance can result in long running jobs for jobs with a large number of training documents.',
+          }
+        ),
+        status: VALIDATION_STATUS.WARNING,
+        heading: i18n.translate('xpack.ml.models.dfaValidation.messages.featureImportanceHeading', {
+          defaultMessage: 'Feature importance',
+        }),
+      });
+    }
+
+    if (
+      (topClasses === ALL_CATEGORIES &&
+        depVarCardinality &&
+        depVarCardinality > NUM_CATEGORIES_THRESHOLD) ||
+      (topClasses && topClasses > NUM_CATEGORIES_THRESHOLD)
+    ) {
+      messages.push({
```

Comment:
For the success case, where `topClasses` is below the threshold, is there value in showing a `SUCCESS` message to show that the Top classes check was performed and was OK, as you do for most of the other checks?

Reply:
Added a success message in 2d7ccc146f7e1b7e86daa70627c839f7e79e637c